src/xdocs/config_design.xml

```diff
@@ -878,14 +878,14 @@ public class Foo { // OK, only one top-level class
             <tr>
               <td>max</td>
               <td>Specify maximum allowed number of throws statements.</td>
-              <td><a href="property_types.html#integer">Integer</a></td>
+              <td><a href="property_types.html#Integer">Integer</a></td>
```

Comment:
please use exact type from setter.
it is `int`
https://github.com/checkstyle/checkstyle/blob/ae324ed2d8eebcdfc8f164095de11b7ae6714a32/src/main/java/com/puppycrawl/tools/checkstyle/checks/design/ThrowsCountCheck.java#L230

please extend page that have list of all types to reference all types we support. We just should be exact on what module is actually has. We unhide extra detail only for enums to let people know possible values, but still name is exact name of enum type.

Reply:
Setting the exact type of setter will be highly inconsistent, especially for `Pattern`  and `Pattern[]`.
For example, `staticGroups` has filed class `Pattern[]`, but setter type `String...`
https://github.com/checkstyle/checkstyle/blob/ae324ed2d8eebcdfc8f164095de11b7ae6714a32/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java#L416
https://github.com/checkstyle/checkstyle/blob/ae324ed2d8eebcdfc8f164095de11b7ae6714a32/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java#L514

The field Class is more accurate description for a property I guess.

Reply:
We should not go in internals of Checks. Inyernal field could change at any time and it should not change type that users set. 

This `String... packageGroups) ` should be `String[]` as it is array, that split by `,` and later each element converted to Pattern.  If user try to use `,` as part of regexp he will experience problems. 

So it is better to tell user a truth. Even for enums we do `String value of BlockOption` name to make it transperet it is a String and there is possible values in BlockOption

Reply:
it is done.